src/mgr/DaemonServer.cc

```diff
@@ -1147,6 +1147,197 @@ void DaemonServer::_maximize_ok_to_stop_set(
   }
 }
 
+void DaemonServer::_update_upgraded_osds(
+  const std::set<int>& orig_osds,
+  const std::set<int>& to_upgrade,
+  const std::set<int>& upgraded,
+  const std::set<int>& version_unknown,
+  upgrade_osd_report *report)
+{
+  // reset output
+  *report = upgrade_osd_report();
+  report->osds = orig_osds;
+  report->ok_upgrade = to_upgrade;
+  report->ok_upgraded = upgraded;
+  report->bad_no_version = version_unknown;
+}
+
+void DaemonServer::_maximize_ok_to_upgrade_set(
+  const std::set<int>& orig_osds,
+  unsigned max,
```

Comment:
why was `max` delcared as `int64_t` in the first place if we were going to expect unsigned

Reply:
ah of course silly me